src/main/java/ca/on/oicr/pineryreports/reports/impl/ProjectSequencingReport.java

```diff
@@ -0,0 +1,212 @@
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    Set<Option> opts = Sets.newHashSet();
+    
+    opts.add(Option.builder()
+        .longOpt(OPT_PROJECT)
+        .hasArg()
+        .argName("code")
+        .required()
+        .desc("Project to report on (required)")
+        .build());
+    
+    return opts;
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    this.project = cmd.getOptionValue(OPT_PROJECT);
+  }
+
+  @Override
+  public String getTitle() {
+    return project + " Sequencing Report";
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
+    List<SampleDto> samples = pinery.getSample().all();
+    allSamplesById = mapSamplesById(samples);
+    Map<Integer, UserDto> allUsersById = pinery.getUser().all().stream()
+        .collect(Collectors.toMap(UserDto::getId, Functions.identity()));
+    
+    List<ReportObject> rows = new ArrayList<>();
+    List<RunDto> allRuns = pinery.getSequencerRun().all();
+    for (RunDto run : allRuns) {
+      if ("Completed".equals(run.getState()) && run.getPositions() != null) {
+        for (RunDtoPosition pos : run.getPositions()) {
+          if (pos.getSamples() != null) {
+            for (RunDtoSample sam : pos.getSamples()) {
+              SampleDto dilution = allSamplesById.get(sam.getId());
+              if (project.equals(dilution.getProjectName())) {
+                UserDto poolCreator = pos.getPoolCreatedById() == null ? null : allUsersById.get(pos.getPoolCreatedById());
+                rows.add(new ReportObject(dilution, run, pos, poolCreator));
+              }
+            }
+          }
+        }
+      }
+    }
+    rows.sort(byPoolDate);
+    reportData = rows;
+  }
+  
+  /**
+   * Sort descending by pool date
+   */
+  private final Comparator<ReportObject> byPoolDate = (o1, o2) -> {
+    String o1Created = removeTime(o1.getLane().getPoolCreated());
+    String o2Created = removeTime(o2.getLane().getPoolCreated());
+    if (o1Created == null) {
+      if (o2Created != null) {
+        return -1;
+      }
+    } else if (o2Created == null) {
+      if (o1Created != null) {
+        return 1;
+      }
+    } else {
+      return o1Created.compareTo(o2Created) * -1;
+    }
+    return 0;
+  };
+
+  @Override
+  protected List<ColumnDefinition> getColumns() {
+    return COLUMNS;
+  }
+
+  @Override
+  protected int getRowCount() {
+    return reportData.size();
+  }
+
+  @Override
+  protected String[] getRow(int rowNum) {
+    ReportObject obj = reportData.get(rowNum);
+    String[] row = new String[COLUMNS.size()];
+    SampleDto stock = getParent(obj.getDilution(), "Stock", allSamplesById);
+    
+    String i1 = getUpstreamAttribute("Barcode", obj.getDilution(), allSamplesById);
+    String i2 = getUpstreamAttribute("Barcode Two", obj.getDilution(), allSamplesById);
+    String indices = i1 == null ? "" : i2 == null ? i1 : i1 + ", " + i2;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)